asm/macros/battle_script.inc

```diff
@@ -0,0 +1,1280 @@
+	.endm
+
+	.macro typecalc2
+	.byte 0x4a
+	.endm
+
+	.macro returnatktoball
+	.byte 0x4b
+	.endm
+
+	.macro getswitchedmondata bank
+	.byte 0x4c
+	.byte \bank
+	.endm
+
+	.macro switchindataupdate bank
+	.byte 0x4d
+	.byte \bank
+	.endm
+
+	.macro switchinanim bank, param1
+	.byte 0x4e
+	.byte \bank
+	.byte \param1
+	.endm
+
+	.macro jumpifcantswitch bank, param1
+	.byte 0x4f
+	.byte \bank
+	.4byte \param1
+	.endm
+
+	.macro openpartyscreen param0, param1
+	.byte 0x50
+	.byte \param0
+	.4byte \param1
+	.endm
+
+	.macro switchhandleorder bank, param1
+	.byte 0x51
+	.byte \bank
+	.byte \param1
+	.endm
+
+	.macro switchineffects bank
+	.byte 0x52
+	.byte \bank
+	.endm
+
+	.macro trainerslidein bank
+	.byte 0x53
+	.byte \bank
+	.endm
+
+	.macro playse param0
+	.byte 0x54
+	.2byte \param0
+	.endm
+
+	.macro fanfare param0
+	.byte 0x55
+	.2byte \param0
+	.endm
+
+	.macro playfaintcry bank
+	.byte 0x56
+	.byte \bank
+	.endm
+
+	.macro atk57
+	.byte 0x57
+	.endm
+
+	.macro returntoball bank
+	.byte 0x58
+	.byte \bank
+	.endm
+
+	.macro handlelearnnewmove param0, param1, param2
+	.byte 0x59
+	.4byte \param0
+	.4byte \param1
+	.byte \param2
+	.endm
+
+	.macro yesnoboxlearnmove param0
+	.byte 0x5a
+	.4byte \param0
+	.endm
+
+	.macro yesnoboxstoplearningmove param0
+	.byte 0x5b
+	.4byte \param0
+	.endm
+
+	.macro hitanimation bank
+	.byte 0x5c
+	.byte \bank
+	.endm
+
+	.macro getmoneyreward
+	.byte 0x5d
+	.endm
+
+	.macro atk5E bank
+	.byte 0x5e
+	.byte \bank
+	.endm
+
+	.macro atk5F
+	.byte 0x5f
+	.endm
+
+	.macro incrementgamestat param0
+	.byte 0x60
+	.byte \param0
+	.endm
+
+	.macro drawpartystatussummary bank
+	.byte 0x61
+	.byte \bank
+	.endm
+
+	.macro atk62 bank
+	.byte 0x62
+	.byte \bank
+	.endm
+
+	.macro jumptorandomattack param0
+	.byte 0x63
+	.byte \param0
+	.endm
+
+	.macro statusanimation bank
+	.byte 0x64
+	.byte \bank
+	.endm
+
+	.macro status2animation bank, status2
+	.byte 0x65
+	.byte \bank
+	.4byte \status2
+	.endm
+
+	.macro chosenstatusanimation bank, param1, param2
+	.byte 0x66
+	.byte \bank
+	.byte \param1
+	.4byte \param2
+	.endm
+
+	.macro yesnobox
+	.byte 0x67
+	.endm
+
+	.macro cancelallactions
+	.byte 0x68
+	.endm
+
+	.macro adjustsetdamage
+	.byte 0x69
+	.endm
+
+	.macro removeitem bank
+	.byte 0x6a
+	.byte \bank
+	.endm
+
+	.macro atknameinbuff1
+	.byte 0x6b
+	.endm
+
+	.macro drawlvlupbox
+	.byte 0x6c
+	.endm
+
+	.macro resetsentmonsvalue
+	.byte 0x6d
+	.endm
+
+	.macro setatktoplayer0
+	.byte 0x6e
+	.endm
+
+	.macro makevisible bank
+	.byte 0x6f
+	.byte \bank
+	.endm
+
+	.macro recordlastability bank
+	.byte 0x70
+	.byte \bank
+	.endm
+
+	.macro buffermovetolearn
+	.byte 0x71
+	.endm
+
+	.macro jumpifplayerran param0
+	.byte 0x72
+	.4byte \param0
+	.endm
+
+	.macro hpthresholds bank
+	.byte 0x73
+	.byte \bank
+	.endm
+
+	.macro hpthresholds2 bank
+	.byte 0x74
+	.byte \bank
+	.endm
+
+	.macro useitemonopponent
+	.byte 0x75
+	.endm
+
+	.macro various bank, param1
+	.byte 0x76
+	.byte \bank
+	.byte \param1
+	.endm
```

Comment:
there should be macros for the `VARIOUS` constants instead of using them directly:
```asm
	.macro cancel_multi_turn_moves bank
	.byte 0x76
	.byte \bank
	.byte VARIOUS_CANCEL_MULTI_TURN_MOVES
	.endm
```